pkg/agent/packetcapture/packetcapture_controller.go

```diff
@@ -567,119 +579,111 @@ func (c *Controller) uploadPackets(ctx context.Context, pc *crdv1alpha1.PacketCa
+			}
 		} else {
-			conditions = append(conditions, crdv1alpha1.PacketCaptureCondition{
-				Type:               crdv1alpha1.PacketCaptureComplete,
-				Status:             metav1.ConditionStatus(v1.ConditionFalse),
-				LastTransitionTime: metav1.Now(),
-				Reason:             "CaptureFailed",
-				Message:            state.err.Error(),
-			})
-		}
-		if toUpdate.Spec.FileServer != nil && state.filePath != "" {
-			conditions = append(conditions, crdv1alpha1.PacketCaptureCondition{
-				Type:               crdv1alpha1.PacketCaptureFileUploaded,
+			conditionStarted = crdv1alpha1.PacketCaptureCondition{
+				Type:               crdv1alpha1.PacketCaptureStarted,
 				Status:             metav1.ConditionStatus(v1.ConditionFalse),
 				LastTransitionTime: t,
-				Reason:             "UploadFailed",
-				Message:            state.err.Error(),
-			})
-		}
-		if state.phase == packetCapturePhasePending {
-			conditions = []crdv1alpha1.PacketCaptureCondition{
-				{
-					Type:               crdv1alpha1.PacketCaptureStarted,
-					Status:             metav1.ConditionStatus(v1.ConditionFalse),
-					LastTransitionTime: t,
-					Reason:             "StartFailed",
-					Message:            state.err.Error(),
-				},
+				Reason:             "Pending",
 			}
 		}
-	} else {
-		if state.isCaptureSuccessful() {
-			conditions = []crdv1alpha1.PacketCaptureCondition{
-				{
-					Type:               crdv1alpha1.PacketCaptureComplete,
-					Status:             metav1.ConditionStatus(v1.ConditionTrue),
-					LastTransitionTime: t,
-					Reason:             "Succeed",
-				},
+		conditions = append(conditions, conditionStarted)
+	case packetCapturePhaseStarted:
+		conditionStarted = crdv1alpha1.PacketCaptureCondition{
+			Type:               crdv1alpha1.PacketCaptureStarted,
+			Status:             metav1.ConditionStatus(v1.ConditionTrue),
+			LastTransitionTime: t,
+			Reason:             "Started",
+		}
+		conditionComplete = crdv1alpha1.PacketCaptureCondition{
+			Type:               crdv1alpha1.PacketCaptureComplete,
+			Status:             metav1.ConditionStatus(v1.ConditionFalse),
+			LastTransitionTime: t,
+			Reason:             "Progressing",
+		}
+		conditions = append(conditions, conditionStarted, conditionComplete)
+	case packetCapturePhaseComplete:
+		conditionStarted = crdv1alpha1.PacketCaptureCondition{
+			Type:               crdv1alpha1.PacketCaptureStarted,
+			Status:             metav1.ConditionStatus(v1.ConditionTrue),
+			LastTransitionTime: t,
+			Reason:             "Started",
+		}
+		reason := "Succeed"
+		message := ""
+		if state.captureErr != nil {
+			if errors.Is(state.captureErr, context.DeadlineExceeded) {
+				reason = "Timeout"
+			} else {
+				reason = "Failed"
 			}
-			if toUpdate.Spec.FileServer != nil {
-				conditions = append(conditions, crdv1alpha1.PacketCaptureCondition{
+			message = state.captureErr.Error()
+		}
+		conditionComplete = crdv1alpha1.PacketCaptureCondition{
```

Comment:
Move `reason := "Succeed"` before this line should be more proper.

Reply:
Ignore this one, I realized this line also initialized the var. So just leave it.